core/src/main/scala/io/finch/request/package.scala

```diff
@@ -309,7 +358,9 @@ package object request extends LowPriorityImplicits {
    *
    * @param message the message
    */
-  class RequestReaderError(val message: String) extends Exception(message)
+  class RequestReaderError(val message: String, cause: Throwable) extends Exception(message, cause) {
```

Comment:
Can we wrap `cause` in `Option`? I'm worried about `null`. It's not a big deal but this is probably the first occurrence of `null` in the entire codebase :)